apps/omg_eth/test/omg_eth/root_chain_test.exs

```diff
@@ -0,0 +1,155 @@
+      txhash_contract: hex_to_binary.("0x3d517d431daea71a99310f12468ffdf2bf547ad1d148f42acfc4ee34dd4e84d7"),
+      plasma_framework: hex_to_binary.("0xd17e1233a03affb9092d5109179b43d6a8828607"),
+      eth_vault: hex_to_binary.("0x1967d06b1faba91eaadb1be33b277447ea24fa0e"),
+      erc20_vault: hex_to_binary.("0xaef6182310e3d34b6ea138b60d36a245386f3201"),
+      payment_exit_game: hex_to_binary.("0x902719f192aa5240632f704aa7a94bab61b86550"),
+      authority_address: hex_to_binary.("0x22d491bde2303f2f43325b2108d26f1eaba1e32b")
+    }
+    {:ok, contract: contract}
+  end
+
+  test "get_root_deployment_height/2 returns current block number", %{contract: contract} do
+    use_cassette "ganache/get_root_deployment_height", match_requests_on: [:request_body] do
+      {:ok, number} = RootChain.get_root_deployment_height(contract.txhash_contract, contract)
+      assert is_integer(number)
+    end
+  end
+
+  test "get_next_child_block/1 returns next blknum to be mined by operator", %{contract: contract} do
+    use_cassette "ganache/get_next_child_block", match_requests_on: [:request_body] do
+      assert {:ok, 1000} = RootChain.get_next_child_block(contract)
+    end
+  end
+
+  describe "has_token/2" do
+    # TODO achiurizo
+    #
+    # Figure out why I can't use the same cassettes even though request_body is unique
+    @tag :skip
+    test "returns true  if token exists", %{contract: contract} do
+      use_cassette "ganache/has_token_true", match_requests_on: [:request_body] do
+        assert {:ok, true} = RootChainHelper.has_token(@eth, contract)
+      end
+    end
+
+    # TODO achiurizo
+    #
+    # Skipping these specs for now as this function needs to be updated
+    # to use the new ALD function (not hasToken?)
+    @tag :skip
+    test "returns false if no token exists", %{contract: contract} do
+      use_cassette "ganache/has_token_false", match_requests_on: [:request_body] do
+        assert {:ok, false} = RootChainHelper.has_token(<<1::160>>, contract)
+      end
+    end
+  end
+
+  test "get_child_chain/2 returns the current block hash and timestamp", %{contract: contract} do
+    use_cassette "ganache/get_child_chain", match_requests_on: [:request_body] do
+      {:ok, {child_chain_hash, child_chain_time}} = RootChain.get_child_chain(0, contract)
+
+      assert is_binary(child_chain_hash)
+      assert byte_size(child_chain_hash) == 32
+      assert is_integer(child_chain_time)
+    end
+  end
+
+  test "submit_block/1 submits a block to the contract", %{contract: contract} do
+    use_cassette "ganache/submit_block", match_requests_on: [:request_body] do
+      block =
+        RootChain.submit_block(
+          <<234::256>>,
+          1,
+          20_000_000_000,
+          contract.authority_address,
+          contract
+        )
+
+      assert {:ok, _} = DevHelpers.transact_sync!(block)
+    end
+  end
+
+  # TODO achiurizo
+  #
+  # ganache complaining about invalid output encoding
```

Comment:
which `plasma_contracts` sha are you using?

This test is a bit of salt in the eye - we wanted the test to run through sending a more involved transaction and retrieving logs, and for this we need a properly formated `depositTx` argument.

But we are in a dependency now, so we don't really have access to code that would reliably generate `tx`. 

You can either ping me with the SHA and I'll look into, or you can try doing this:

```
      tx =
        [<<1>>, [], [[<<1>>, contract.authority_address, @eth, 1]], <<0::256>>]
        |> ExRLP.encode()
```

instead. If I'm guessing correctly, you're using contracts from `plasma-contracts/master` while this branches off of `elixir-omg/master` which doesn't yet loop `master` contracts in (pending #1005).

Or alternatively (and even more correctly I think), generate the cassettes with the `plasma-contracts` SHA that `mix.lock` references - it should match.

Reply:
In my opinion, the tests purpose is wrong. it's trying to cover everything. That's stretching what the purpose of a unit test should be. Instead, we should focus on testing the bits and pieces that lead to a successful recognition of a deposit, for example,
`OMG.Eth.get_ethereum_events/4` is untested (has no unit test to be exact!), and if we provide inputs from exvcr we will prove that parsing works and we do not need to focus on transacting and all that setup.

Reply:
It was never the intention for this to be a `unit` test, see https://github.com/omisego/elixir-omg/blob/master/apps/omg_eth/test/omg_eth/eth_test.exs#L17

Reply:
a related note here is that maybe it's time to sunset the guts of OMG.Eth.RootChain in favor of leveraging a proper web3 lib (if there's one that satisfies our needs ofc). It was since long on the list and was even explored a bit a long time ago:

I moved the respective ticket from Jira (or in fact, pivotal tracker) to https://github.com/omisego/elixir-omg/issues/1041

Reply:
Locked this to dc3d60a72b814916902c343cea30e65f1e4144c3 from `plasma-contracts`. See f9435faf87a881c8a3a9f83de76b411ce8c569a2

Reply:
that contracts' SHA seems to be recent. If you're merged with `elixir-omg-master`, doing the `<<1>>` modification I mentioned should work (in fact, it should be modified in `master`, possibly lost during the merge)